### PR TITLE
Scripts for generating values report and NIDM jsonld files

### PR DESCRIPTION
**Pull request by zxenia**
## Description

Scripts that generate:
1. Report about distinct values in each of the following properties: `privacy`, `types`, `licenses`, `isAbout`, `formats`, `keywords`.

2. Duplicates file that contains string matching values spelled in different cases (e.g. lowercases vs uppercase  vs title)

3. JSON-LD file for each distinct value and annotate value with an Interlex term match (using [NIF API](https://scicrunch.org/api/1/ilx/search/term/)). By default creates a directory for each of the above properties and saves jsonld files in a respective directory.

### CODE REVIEW

**Comment by cmadjar** [comment]
@zxenia is this ready for review? If not, just tag me when ready. Just want to be sure you are not waiting after me.

FYI - the Circle CI failure should be fixed once #530 is be merged. You will just need to merge conp/master once the PR #530 is merged.

**Comment by zxenia** [comment]
@cmadjar Yes, it's ready for review. I will merge the master when 530 is merged. Thanks!

File: scripts/conp_to_nidm_terms/functions.py
Comment (cmadjar) [inline scripts/conp_to_nidm_terms/functions.py:61]:
```suggestion
    Iterates over the projects directory content retrieving DATS file for each project.
```

File: scripts/conp_to_nidm_terms/functions.py
Comment (cmadjar) [inline scripts/conp_to_nidm_terms/functions.py:63]:
```suggestion
    :param : set to False in order to exclude the property from the final report
```

File: scripts/conp_to_nidm_terms/functions.py
Comment (cmadjar) [inline scripts/conp_to_nidm_terms/functions.py:105]: I'm confused. IsAbout can have both `name` and `value`? Is that expected?

Also, in here, you could maybe could skip the terms that have an identifier, like species who are related to NCBI taxonomy terms?

File: scripts/conp_to_nidm_terms/functions.py
Comment (cmadjar) [inline scripts/conp_to_nidm_terms/functions.py:120]: cool! I learned something new. Did not know the `zip` function. Awesome!

File: scripts/conp_to_nidm_terms/functions.py
Comment (cmadjar) [inline scripts/conp_to_nidm_terms/functions.py:133]:
```suggestion
    Finds duplicate values spelled in different cases (e.g. lowercase vs uppercase vs title)
```

File: scripts/conp_to_nidm_terms/functions.py
Comment (cmadjar) [inline scripts/conp_to_nidm_terms/functions.py:159]:
```suggestion
    Generates a JSON-LD file for each unique term.
```

File: scripts/conp_to_nidm_terms/functions.py
Comment (cmadjar) [inline scripts/conp_to_nidm_terms/functions.py:160]:
```suggestion
    Files are saved to the directories respective to their properties.
```

I think? Not sure...

File: scripts/conp_to_nidm_terms/functions.py
Comment (cmadjar) [inline scripts/conp_to_nidm_terms/functions.py:177]:
```suggestion
            # Create and save JSON-LD file in the respective folder
```

File: scripts/conp_to_nidm_terms/report_generator.py
Comment (cmadjar) [inline scripts/conp_to_nidm_terms/report_generator.py:38]: might be worth adding a print statement here to say that duplicates were found and that the duplicates.txt file has been generated.

File: scripts/conp_to_nidm_terms/jsonld_generator.py
Comment (cmadjar) [inline scripts/conp_to_nidm_terms/jsonld_generator.py:22]: could maybe add a check here to see if the api_key has been set so that we do not have to wait after the report is created to know that we did not set the key.

Currently, if I run the script without having set the api_key, I get the following:
```
DATS files processed: 51
Traceback (most recent call last):
  File "jsonld_generator.py", line 43, in <module>
    main(argv[1:])
  File "jsonld_generator.py", line 33, in main
    generate_jsonld_files(report=report, use_api=use_api)
  File "/Users/cmadjar/GitHub/CONP-PCNO/conp-dataset/scripts/conp_to_nidm_terms/functions.py", line 172, in generate_jsonld_files
    jsonld_description["sameAs"] = get_api_response(term.lower())
  File "/Users/cmadjar/GitHub/CONP-PCNO/conp-dataset/scripts/conp_to_nidm_terms/functions.py", line 38, in get_api_response
    key = api_key_json["api_key"] if api_key_json["api_key"] != "" else _raise_error(f"{api_key_json['_comment']}")
  File "/Users/cmadjar/GitHub/CONP-PCNO/conp-dataset/scripts/conp_to_nidm_terms/functions.py", line 25, in _raise_error
    def _raise_error(er): raise Exception(er)
Exception: Add your API Key for the NIF data services above. More information on how to obtain API Key can be found here https://neuinfo.org/about/webservices
```

**Review by cmadjar — CHANGES_REQUESTED** [review]
Looks great! A few comments.

Note: there is a weird bug at the moment in Github where when you click on the `suggestion` button, it inserts twice the `suggestion` bits. Just letting you know in case I forgot to remove the duplicate instance that has no suggestion.

Thank you!!

**Comment by cmadjar** [comment]
@zxenia the PR was merged into master with the updates that will resolve the Circle CI issue of this PR. You can now merge master into your branch to update your branch with the latest code.

File: scripts/conp_to_nidm_terms/functions.py
Comment (zxenia) [reply scripts/conp_to_nidm_terms/functions.py:25]: I raise the error below in the one-line conditional expression and it cannot take another statement (raise) only its result

**Comment by zxenia** [comment]
Hi @cmadjar , I've implemented your comments.
The CircleCI tests are still failing after merging the master, it seems to be related to the tests code.

File: scripts/conp_to_nidm_terms/functions.py
Comment (zxenia) [reply scripts/conp_to_nidm_terms/functions.py:106]: `isAbout` can take nine different schemas inside, all of them have `name `required except for annotation_schema - the string value is in `value` there, so I am checking for it.


> Also, in here, you could maybe could skip the terms that have an identifier, like species who are related to NCBI taxonomy terms?

Do you mean the script shouldn't create a jsonld file for the species terms?

Comment (cmadjar) [reply scripts/conp_to_nidm_terms/functions.py:106]: @zxenia I don't know. Do the species need to be added to the NIDM terms if they are terms coming from the NCBI taxonomy database? I honestly don't know.

**Review by cmadjar — APPROVED** [review]
LGTM

**Comment by cmadjar** [comment]
@mathdugre @joeyzhou98  Looks like the Circle CI error here is related to flake8. @zxenia merged the latest version of the master branch into her branch so I am not sure why this is failing. Any idea?

**Comment by cmadjar** [comment]
Actually, I think I found the answer. It might be fixed by this once merged: https://github.com/CONP-PCNO/conp-dataset/pull/550

**Comment by zxenia** [comment]
Ok, I will merge it when #550 merged, thank you!

**Comment by zxenia** [comment]
@cmadjar I have merged the latest master changes, now test build still fails due to the download fail for 1000Genomes project:
```
[ERROR] download failed
| download failed
| download failed [get(/home/circleci/conp-dataset/projects/1000GenomesProject/1KGP_chrMT.vcf.gz)] 
```

**Comment by cmadjar** [comment]
Manual download of that file worked. This might have been just a transient issue. I restarted the test where it failed. Let's see what it gives.

**Comment by cmadjar** [comment]
@zxenia Emmet is working on fixing this today. Once the PR to resolve that is merged, I will notify you and you will need to merge again conp-dataset/master into your branch. Hopefully after that the tests will pass :). Thank you!

**Comment by zxenia** [comment]
@cmadjar Sounds good, thank you very much!

**Comment by cmadjar** [comment]
@zxenia ready to merge conp-dataset/master into your branch. The dataset has been updated.

**Comment by zxenia** [comment]
Yassss, it passed now!
@cmadjar thanks a lot! could you please merge it, I don't have write access.

**Pull request merged**
